tests/cypress/tests/unit/syntactic_tree_test.ts

```diff
@@ -1,10 +1,92 @@
+
+            expect(tree.isValid).to.equal(true);
+            expect(tree.evaluate(data0)).to.equal(false);
+            expect(tree.evaluate(data1)).to.equal(true);
+            expect(tree.evaluate(data2)).to.equal(true);
+            expect(tree.evaluate(data3)).to.equal(false);
+        });
+
+        it('support column name with "-"', () => {
+            const tree = new SyntaxTree('a-dot eq "1-dot" || a-dot eq "2-dot"');
+
+            expect(tree.isValid).to.equal(true);
+            expect(tree.evaluate(data0)).to.equal(false);
+            expect(tree.evaluate(data1)).to.equal(true);
+            expect(tree.evaluate(data2)).to.equal(true);
+            expect(tree.evaluate(data3)).to.equal(false);
+        });
+
+        it('support column name with "_"', () => {
+            const tree = new SyntaxTree('a_dot eq "1_dot" || a_dot eq "2_dot"');
+
+            expect(tree.isValid).to.equal(true);
+            expect(tree.evaluate(data0)).to.equal(false);
+            expect(tree.evaluate(data1)).to.equal(true);
+            expect(tree.evaluate(data2)).to.equal(true);
+            expect(tree.evaluate(data3)).to.equal(false);
+        });
+
+        it('support column name with "+"', () => {
+            const tree = new SyntaxTree('a+dot eq "1+dot" || a+dot eq "2+dot"');
+
+            expect(tree.isValid).to.equal(true);
+            expect(tree.evaluate(data0)).to.equal(false);
+            expect(tree.evaluate(data1)).to.equal(true);
+            expect(tree.evaluate(data2)).to.equal(true);
+            expect(tree.evaluate(data3)).to.equal(false);
+        });
+
+        it('support column name with ":"', () => {
+            const tree = new SyntaxTree('a:dot eq "1:dot" || a:dot eq "2:dot"');
+
+            expect(tree.isValid).to.equal(true);
+            expect(tree.evaluate(data0)).to.equal(false);
+            expect(tree.evaluate(data1)).to.equal(true);
+            expect(tree.evaluate(data2)).to.equal(true);
+            expect(tree.evaluate(data3)).to.equal(false);
+        });
+
+        it('support double quoted column name with " " (space)', () => {
+            const tree = new SyntaxTree('"a dot" eq "1 dot" || "a dot" eq "2 dot"');
+
+            expect(tree.isValid).to.equal(true);
+            expect(tree.evaluate(data0)).to.equal(false);
+            expect(tree.evaluate(data1)).to.equal(true);
+            expect(tree.evaluate(data2)).to.equal(true);
+            expect(tree.evaluate(data3)).to.equal(false);
+        });
+
+        it('support single quoted column name with " " (space)', () => {
+            const tree = new SyntaxTree('\'a dot\' eq "1 dot" || \'a dot\' eq "2 dot"');
+
+            expect(tree.isValid).to.equal(true);
+            expect(tree.evaluate(data0)).to.equal(false);
+            expect(tree.evaluate(data1)).to.equal(true);
+            expect(tree.evaluate(data2)).to.equal(true);
+            expect(tree.evaluate(data3)).to.equal(false);
+        });
+    });
```

Comment:
Unit tests for the various special cases

Reply:
Can you include some nested quotes?

Reply:
Added